modules/workflow-service-api/src/main/java/org/opencastproject/workflow/api/AbstractWorkflowOperationHandler.java

```diff
@@ -360,6 +378,160 @@ protected String getConfig(WorkflowOperationInstance woi, String key) throws Wor
+   */
+  protected ConfiguredTagsAndFlavors getTagsAndFlavors(WorkflowInstance wi, Configuration srcTags, Configuration srcFlavors, Configuration targetTags, Configuration targetFlavors) throws WorkflowOperationException {
+    WorkflowOperationInstance woi = wi.getCurrentOperation();
+    ConfiguredTagsAndFlavors tagsAndFlavors = new ConfiguredTagsAndFlavors();
+    MediaPackageElementFlavor flavor;
+
+    List<String> srcTagList = new ArrayList<>();
+    String srcTag;
+    switch(srcTags) {
+      case none:
+        break;
+      case one:
+        srcTag = StringUtils.trimToNull(woi.getConfiguration(SOURCE_TAG));
+        if (srcTag == null) {
+          throw new WorkflowOperationException("Configuration key '" + SOURCE_TAG + "' must be set");
+        } else {
+          srcTagList.add(srcTag);
+        }
+        break;
+      case many:
+        srcTagList = asList(StringUtils.trimToNull(woi.getConfiguration(SOURCE_TAGS)));
+        srcTag = StringUtils.trimToNull(woi.getConfiguration(SOURCE_TAG));
+        if (srcTagList.isEmpty() && srcTag != null) {
+          srcTagList.add(srcTag);
+        }
+        if (srcTagList.isEmpty()) {
+          logger.warn("Configuration keys '" + SOURCE_TAGS + "and" + SOURCE_TAG + "' are empty");
+        }
+        break;
+      default:
+        throw new WorkflowOperationException("Couldn't process srcTags configuration option!");
+    }
+    tagsAndFlavors.setSrcTags(srcTagList);
+
+    List<String> srcFlavorList = new ArrayList<>();
+    String sourceFlavor;
+    switch(srcFlavors) {
+      case none:
+        break;
+      case one:
+        sourceFlavor = StringUtils.trimToNull(woi.getConfiguration(SOURCE_FLAVOR));
+        if (sourceFlavor == null) {
+          throw new WorkflowOperationException("Configuration key '" + SOURCE_FLAVOR + "' must be set");
+        } else {
+          srcFlavorList.add(sourceFlavor);
+          try {
+            flavor = MediaPackageElementFlavor.parseFlavor(srcFlavorList.get(0));
+          } catch (IllegalArgumentException e) {
+            throw new WorkflowOperationException(srcFlavorList.get(0) + " is not a valid flavor!");
+          }
+        }
+        break;
+      case many:
+        srcFlavorList = asList(StringUtils.trimToNull(woi.getConfiguration(SOURCE_FLAVORS)));
+        sourceFlavor = StringUtils.trimToNull(woi.getConfiguration(SOURCE_FLAVOR));
+        if (srcFlavorList.isEmpty() && sourceFlavor != null) {
+          srcFlavorList.add(sourceFlavor);
+        }
+        if (srcFlavorList.isEmpty()) {
+          logger.warn("Configuration keys '" + SOURCE_FLAVORS + "and" + SOURCE_FLAVOR + "' are empty");
+        } else {
+          for (String elem : srcFlavorList) {
+            try {
+              flavor = MediaPackageElementFlavor.parseFlavor(elem);
+            } catch (IllegalArgumentException e) {
+              throw new WorkflowOperationException(elem + " is not a valid flavor!");
+            }
+          }
+        }
```

Comment:
No need to check if the list is empty. It won't loop over anything if it is.
```suggestion
        for (String elem : srcFlavorList) {
          try {
            flavor = MediaPackageElementFlavor.parseFlavor(elem);
          } catch (IllegalArgumentException e) {
            throw new WorkflowOperationException(elem + " is not a valid flavor!");
          }
        }
```